releases/stable-branches-and-release-cycle.md

```diff
@@ -0,0 +1,79 @@
+# Producing and supporting stable releases
+
+## Summary
+
+Today the Submariner project releases one minor version a month without capacity
+for stable branches or bugfix releases. In the following proposal we define a plan
+to make this possible starting with 0.8.
+
+## Proposal
+
+### Alignment of branch -> image releases
+
+We produce currently `<image>:devel` for every master push, along with `<image>:x.y.z` for
+the releases.
+
+Changing the `master` branch name to `devel` would make this homogeneous so our image releases
+are simply `<image>:<branch-or-tag>`.
+
+### Creating one stable branch for every minor release
+
+Approaching the release of every minor or major version a new stable branch will be forked
+off `devel` branch in the form of `release-x.y` for every existing repository.
+
+Any changes related to pinning shipyard base image will be done in that branch.
+
+At that point in time, any patches which belong to the release will need to be merged in master
+and backported to `release-x.y`. For this reason, the later we branch, the less extra work
+will be necessary to produce a stable release.  
+
+### Maintaining a stable branch
+
+Any patches that are necessary for fixing bugs in a stable branch will need to be merged
+first into master, and the later stable branches.
+
+`release-0.8`  <-- `release-0.9` <-- `devel`
+
+Patches could be simple cherry-picks, or sometimes they could need specific adaptations
+to the branch.
+
+When a patch is merged to a stable branch the images for such branch are updated, for example
+`quay.io/submariner/submariner:release-0.9`
+
+### How long is a stable branch maintained
+
+Initially stable branches will be maintained until the next stable release is released,
+this would change in the future if we introduce the concept of long term support releases.
+
+### Reducing the cadence of minor releases
+
+Based on the previous ideas the existing minor version cadence doesn't seem realistic
+for maintaining stable branches. We don't want fake stable branches where patches
+are never backported.
+
+With the new cadence after each sprint we would produce a milestone release,
+for example
+
+* `0.9-milestone-1-rc0`, ... , `0.9-milestone-1`
+* `0.9-milestone-2-rc0`, ... , `0.9-milestone-2`
+* `0.9-milestone-3-rc0`, ... , `0.9-milestone-3`
+* `0.9-milestone-4-rc0`, ... , `0.9-milestone-4`
+
+In this case we won't do pinning of shipyard base image, but we will release and update
```

Comment:
I think this can apply to stable branches as well, e.g. when we create the branch we'll pin it to shipyard 'release-x.y.z' which will always remain on the compatible version.
Makes sense?

Reply:
L-25 you mean, or something different?

Reply:
No I mean that specifically each stable branch will be pinned to the corresponding shipyard version, but we won't be pinning to a fixed version.

So for example, `release-0.8` branch on `submariner` will be pinned to `shipyard-dapper-base:release-0.8` and won't be changed when releasing (as it makes less sense if we're already on a stable branch).

Reply:
Yes, but in this paragraph I'm talking about the milestones, right?, on the milestones we don't pin. On the RC's yes. I'll clarify that.

Reply:
Ah, I think I see what you mean, but go.mod can't be pointed to "release branches", only to vx.x.x tags, or git shas..

Reply:
I'm talking about the 'Dockerfile.dapper' files only, not 'go.mod'

Reply:
> I'm talking about the 'Dockerfile.dapper' files only, not 'go.mod'

Ack, still not understanding it :), when you have the time specific text changes would help me I guess ':D sorry, I'm a bit thick today...